### ISSUE TITLE
Framebuffer is rendered to memory too early in Danganronpa's MTBs

### ISSUE THREAD

**Issue by internetakias** [hrydgard/ppsspp#8708]
After you defeat an opponent in a Machine gun Talk Battle, the game shows different shots of the opponent, and then the screen "breaks".

Now when the screen breaks, the second to last shot is displayed, instead of the last one.
What happens in PPSSPP:  http://y.zxq.co/znkkhn.webm
What should happen: https://youtu.be/lq4VrkVVCMI?t=1m
This happens in both "Render to memory" modes. Normal buffered rendering is broken.

**Comment by internetakias** [comment]
Frame dump of when this happens
http://pastebin.com/dK83zGPT

**Comment by unknownbrackets** [comment]
This is most likely because of the async download, which only happens in OpenGL.

Does this happen in Vulkan / Direct3D 11?

-[Unknown]

**Comment by internetakias** [comment]
I'm gonna check right now

**Comment by internetakias** [comment]
Was Render to Memory removed? I can't find it, and Buffered render with Block Transfer Effects isn't rendering the framebuffers properly at all.

**Comment by unknownbrackets** [comment]
It was removed - it generally made games crash, or at best just made them slower.  We should already detect the block transfers in Danganronpa, I thought...

So now the shatter thing doesn't work?  Can you use the GE debugger on Windows to see where it renders the picture of the girl, before the shatter effect (the one that should be downloaded to RAM?)

https://github.com/hrydgard/ppsspp/wiki/How-to-find-a-graphic-issue-with-the-GE-debugger

What I'm specifically after here is the framebuffer address, and then we can see why it's not being downloaded.

-[Unknown]

**Comment by internetakias** [comment]
No, it doesn't work and some other effects don't work either now. Like the Load Screen being laid over the title screen, for example. I'll go use that debugger now.

**Comment by internetakias** [comment]
Here's the load screen background framebuffer when you first run the game: 
![image](https://user-images.githubusercontent.com/5077063/34451070-81ef1d5e-ed23-11e7-971a-033a9b0ad686.png)
And here it is again after going to the load screen again.
![image](https://user-images.githubusercontent.com/5077063/34451059-1ed03668-ed23-11e7-88ae-e273bc53494f.png)
You'll notice some text behind the load screen. That's the previous load screen, and it's being used as a framebuffer instead of the black background, or the title screen like it should:
![image](https://user-images.githubusercontent.com/5077063/34451081-e449961e-ed23-11e7-9697-8d6098d5c154.png)

And here's what the debugger shows: https://imgur.com/a/9aqyI ("Watch" was empty)

**Comment by internetakias** [comment]
Here's the MTB: https://imgur.com/a/eH6zQ (Watch was empty again)

**Comment by unknownbrackets** [comment]
If it has an old version of the frame in it, that seems odd - as if the old behavior is still there.

Can you try this with the latest git build, and put it in a clean directory so it has default settings?  You can copy the savedata.

Note: it should be "Buffered rendering" and "Simulate Block Transfer Effects" (which is basically the name of the new, safer version of the old functionality) must be on.  Replacements must also be on, which can't be disabled in the UI, but can be disabled in the ini manually.

All of the settings that need to be on are on by default, so that's why I want to validate that it happens with defaults.

-[Unknown]

**Comment by internetakias** [comment]
I'm using g2dda2bfa7 with Buffered Rendering and Simulate Block Transfer Effects on but I can try with a clean version

**Comment by internetakias** [comment]
Here's the load screen in the clean version: https://imgur.com/a/KGOOk

**Comment by unknownbrackets** [comment]
Is this still an issue in the latest git?  Make sure there are no speed hack settings enabled.

Also, I assume it works just fine using software rendering, right?  It might be a bit slow, but that doesn't need any block transfer simulation.

-[Unknown]

**Comment by Gamemulatorer** [comment]
No feedback/outdated?
